### ISSUE TITLE
the thread safe for accessing array with index, which may happend during incremental decoding

### ISSUE THREAD

**Issue by 342261733** [SDWebImage/SDWebImage#3407]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Sporadic crash occurs online, stack is as shown in the figure below:
![image](https://user-images.githubusercontent.com/8207026/191733480-bd4e78d0-5957-4111-a2ae-9389a92818aa.png)

This issue has been fiexd in SDWebImageWebPCoder:
https://github.com/SDWebImage/SDWebImageWebPCoder/pull/68/files
Remember to update
